DB/api/db.py

```diff
@@ -43,18 +43,27 @@ def query(parsed_url):
 		device = parsed_url['device']
 	if 'diff' in parsed_url.keys():
 		diff = parsed_url['diff']
-
-	results = retrieve_within_filters(
-		device_id,
-		start_time,
-		end_time,
-		data_type,
-		subset,
-		limit,
-		reverse,
-		device,
-		diff,
-	)
+	if parsed_url['granularity']:
```

Comment:
I think its better because all the params are either None or False this way you dont have to do a scan over the list of params every time, not that it matters that much. I'll update.